test/wallet-rpc-test.js

```diff
@@ -0,0 +1,287 @@
+        assert.equal(info.watchOnly, false);
+      };
+    });
+
+    // The rpc interface requires the wallet to be selected first
+    // the "standardWalletId" is not watch only, and the
+    // "watchOnlyWalletId" is watch only. Select both and assert that
+    // the watch only property that is returned properly matches
+    it('should return iswatchonly correctly', async () => {
+      // m/44'/1'/0'/0/0
+      const receive = 'mkpZhYtJu2r87Js3pDiWJDmPte2NRZ8bJV';
+
+      {
+        await wclient.execute('selectwallet', [standardWalletId]);
+        const response = await wclient.execute('getaddressinfo', [receive]);
+        assert.equal(response.iswatchonly, false);
+      }
+      {
+        await wclient.execute('selectwallet', [watchOnlyWalletId]);
+        const response = await wclient.execute('getaddressinfo', [receive]);
+        assert.equal(response.iswatchonly, true);
+      }
+    });
+
+    // Assert that getaddressinfo returns the same address.
+    // The gotcha would be returning an address with the incorrect
+    // hrp which would be a problem with the network the address
+    // object uses
+    it('should return the correct address', async () => {
+      // m/44'/1'/0'/0/0
+      const receive = 'mkpZhYtJu2r87Js3pDiWJDmPte2NRZ8bJV';
+
+      await wclient.execute('selectwallet', [watchOnlyWalletId]);
+      const response = await wclient.execute('getaddressinfo', [receive]);
+      assert.equal(response.address, receive);
+    });
+
+    // Select the watch only wallet that was created with the xpub
+    // at account index 0', derive the same address locally
+    // and pass to getaddressinfo, assert that ismine is true,
+    // select the standard wallet that has a different hd seed
+    // and then pass the same address to getaddressinfo and
+    // assert that ismine is false
+    it('should detect owned address', async () => {
+      // m/44'/1'/0'/0/0
+      const receive = 'mkpZhYtJu2r87Js3pDiWJDmPte2NRZ8bJV';
+      {
+        await wclient.execute('selectwallet', [watchOnlyWalletId]);
+        const response = await wclient.execute('getaddressinfo', [receive]);
+        assert.equal(response.ismine, true);
+      }
+      {
+        await wclient.execute('selectwallet', [standardWalletId]);
+        const response = await wclient.execute('getaddressinfo', [receive]);
+        assert.equal(response.ismine, false);
+      }
+    });
+
+    it('should detect a p2sh address', async () => {
+      const script = Script.fromMultisig(2, 2, pubkeys);
+
+      const address = Address.fromScript(script);
+      const addr = address.toString(network);
+      const response = await wclient.execute('getaddressinfo', [addr]);
+      assert.equal(response.isscript, true);
+      assert.equal(response.iswitness, false);
+      assert.equal(response.witness_program, undefined);
+    });
+
+    // assert that the correct witness program is returned for a
```

Comment:
You could probably add the bit about p2wpkh in the it statement, otherwise the code is clear, and comment not necessary.